pkg/cloud/aws/services/ec2/bastion.go

```diff
@@ -38,7 +38,7 @@ const (
 
 // ReconcileBastion ensures a bastion is created for the cluster
 func (s *Service) ReconcileBastion() error {
-	if s.scope.VPC().IsProvided() {
+	if s.scope.VPC().IsProvided(s.scope.Name()) {
```

Comment:
Is there any way to avoid the repetition here?

Reply:
What did you have in mind? This happened because the "do I own this?" question now implicitly depends on the name of the cluster, and the `VPCSpec` doesn't know its own cluster name.